core/src/apresolve.rs

```diff
@@ -69,7 +88,7 @@ impl ApResolver {
     pub async fn try_apresolve(&self) -> Result<ApResolveData, Error> {
         let req = Request::builder()
             .method(Method::GET)
-            .uri("http://apresolve.spotify.com/?type=accesspoint&type=dealer&type=spclient")
+            .uri("https://apresolve.spotify.com/?type=accesspoint&type=dealer&type=spclient")
```

Comment:
This was not HTTPS because the official client uses HTTP. As long as HTTPS works, that's fine by me, but no guarantees about what Spotify will do in the future.

Reply:
That is a good point then, I was thinking about possible security implications with MITM attacks that could inject malicious APs. I don't know enough about the proceeding protocol to tell if there a malicious AP could do any damage though.

That being said, while the https url definitely works, the original client using HTTP is a somewhat convincing argument. 
On the other hand if Spotify would change anything in the future, I doubt that they would disable secure HTTPS and only allow insecure HTTP. If anything I'd assume the opposite.

Let me know what you think, if I should keep it or get rid of the HTTPS again

Reply:
I am fine with keeping it on HTTPS. If they ever switch that off, the fallback will still work. I think your argument is compelling that it would be strange to turn off something which is secure, but hey, who knows.

I'm not so afraid for MITM attacks because we verify the Spotify server certificate during the handshake. So any malicious access points would be rejected on connection.